src/core_modules/capture-core/components/WidgetStagesAndEvents/Stages/Stage/StageCreateNewButton/StageCreateNewButton.js

```diff
@@ -0,0 +1,72 @@
+// @flow
+import React, { useMemo } from 'react';
+import { Button, IconAdd16 } from '@dhis2/ui';
+import i18n from '@dhis2/d2-i18n';
+import { ConditionalTooltip } from '../../../../Tooltips/ConditionalTooltip';
+
+type Props = {
+    onCreateNew: (stageId: string) => void,
+    stageWriteAccess: ?boolean,
+    eventCount: number,
+    repeatable: ?boolean,
+    preventAddingEventActionInEffect: ?boolean,
+    eventName: string,
+}
+
+export const StageCreateNewButton = ({
+    onCreateNew,
+    stageWriteAccess,
+    eventCount,
+    repeatable,
+    preventAddingEventActionInEffect,
+    eventName,
+}: Props) => {
+    const isDisabled = useMemo(() => {
```

Comment:
Do the two useMemo perform the same checks? If so, how about combining them into only one hook? 
```
const { isDisabled , tooltipContent} = useMemo(() => { 
    ....
    ....
}, [eventCount, eventName, preventAddingEventActionInEffect, repeatable, stageWriteAccess]);
```
Thanks!